streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -110,7 +110,10 @@ public void assignActiveTasks(final Collection<TaskId> tasks) {
     }
 
     public void assignActiveToConsumer(final TaskId task, final String consumer) {
-        assignedActiveTasks.consumerToTaskIds().computeIfAbsent(consumer, k -> new HashSet<>()).add(task);
+        if (assignedActiveTasks.taskIds().contains(task)) {
+            assignedActiveTasks.consumerToTaskIds()
+                               .computeIfAbsent(consumer, k -> new HashSet<>()).add(task);
```

Comment:
If the condition does not satisfy, then we should throw an illegal state exception indicating a bug.